src/skin.cpp

```diff
@@ -52,6 +52,36 @@ Polygons SkinInfillAreaComputation::getInsidePolygons(const SliceLayerPart& part
  * This function is executed in a parallel region based on layer_nr.
  * When modifying make sure any changes does not introduce data races.
  *
+ * this function may only read/write the skin and infill from the *current* layer.
+ */
+Polygons SkinInfillAreaComputation::getWalls(const SliceLayerPart& part_here, int layer2_nr, unsigned int wall_idx)
+{
+    Polygons result;
+    if (layer2_nr < static_cast<int>(mesh.layers.size()))
```

Comment:
This is an input check. You could save an indentation by flipping this if-statement around:

    if (layer2_nr >= static_cast<int>(mesh.layers.size()))
    {
        return Polygons(); //The other layer is above the model, so it must be empty.
    }
    const SliceLayer& layer2...

(Not blocking.)